### ISSUE TITLE
Containers still running, but not available through localhost port

### ISSUE THREAD

**Issue by TheFox** [docker/libnetwork#2185]
Please see [this comment](https://github.com/docker/for-mac/issues/1622#issuecomment-396639352).

**Comment by fcrisciani** [comment]
@TheFox, did you try to force the use of IPv4 with `curl -4 <url>`

**Comment by TheFox** [comment]
I didn't try `curl -4 127.0.0.1:32000` but `telnet 127.0.0.1 32000`. But I'll try. Do you guess this is an issue with IPv6?

**Comment by fcrisciani** [comment]
@TheFox sometimes depending on the OS config, ipv6 is prioritized on ipv4 so curl tries ipv6 but timeout. just an idea to avoid chasing ghosts :)

**Comment by TheFox** [comment]
@fcrisciani I'm sure that IPv4 is used. Since I use the IP address `127.0.0.1` in the reverse proxy no IPv6 will be used.

**Comment by fcrisciani** [comment]
@TheFox does the container ever restart and the failure is maybe happening after that?

**Comment by TheFox** [comment]
@fcrisciani Nope. The container does not restart ever. But when I manually restart the container it works fine again. Restarting the container or even the Docker daemon is no acceptable solution.

I commented to [after ±20 hours: containers still running, but not available through port)(https://github.com/docker/for-mac/issues/1622) in the first place because the description is very similar to my problem. The container is running fine, after ~2 to 3 days (or so, I don't know exactly) the container isn't accessable via the `--publish` port but through the internal Docker Network ip address (from `docker inspect -f '{{ .NetworkSettings.IPAddress }}' CONTAINER_ID`).

What's even the main purpose (or advantage) to have an internal IP address **and** a public when I can use the internal anytime anyway?
